PWGEM/PhotonMeson/TableProducer/gammaSelection.cxx

```diff
@@ -0,0 +1,198 @@
+    hCaloCuts_EMC->GetXaxis()->SetBinLabel(5, "#it{N}_{cell} cut");
+    hCaloCuts_EMC->GetXaxis()->SetBinLabel(6, "TM");
+    hCaloCuts_EMC->GetXaxis()->SetBinLabel(7, "out");
+
+    LOG(info) << "| ECMal cluster cut settings:" << std::endl;
+    LOG(info) << "|\t Timing cut: " << EMC_minTime << " < t < " << EMC_maxTime << std::endl;
+    LOG(info) << "|\t M02 cut: " << EMC_minM02 << " < M02 < " << EMC_maxM02 << std::endl;
+    LOG(info) << "|\t E_min cut: E_cluster > " << EMC_minE << std::endl;
+    LOG(info) << "|\t N_cell cut: N_cell > " << EMC_minNCell << std::endl;
+    LOG(info) << "|\t TM |eta|: |eta| <= " << EMC_TM_Eta->at(0) << " + (pT + " << EMC_TM_Eta->at(1) << ")^" << EMC_TM_Eta->at(2) << std::endl;
+    LOG(info) << "|\t TM |phi|: |phi| <= " << EMC_TM_Phi->at(0) << " + (pT + " << EMC_TM_Phi->at(1) << ")^" << EMC_TM_Phi->at(2) << std::endl;
+    LOG(info) << "|\t TM E/p: E/p < " << EMC_Eoverp << std::endl;
+    LOG(info) << "|\t Cut bit is set to: " << EMC_CutModeBitSet << std::endl;
+
+    gatherCutsEMC(EMC_minTime, EMC_maxTime, EMC_minM02, EMC_maxM02, EMC_minE, EMC_minNCell, EMC_TM_Eta, EMC_TM_Phi, EMC_Eoverp);
+
+    // PHOS
+    PHOSHistos.add("hClusterEIn", "hClusterEIn", gHistoSpec_clusterECuts);
+    PHOSHistos.add("hClusterEOut", "hClusterEOut", gHistoSpec_clusterECuts);
+    auto hCaloCuts_PHOS = PHOSHistos.add<TH1>("hCaloCuts_PHOS", "hCaloCuts_PHOS", kTH1I, {{7, -0.5, 6.5}});
+    hCaloCuts_PHOS->GetXaxis()->SetBinLabel(1, "in");
+    hCaloCuts_PHOS->GetXaxis()->SetBinLabel(2, "#it{t}_{cluster} cut");
+    hCaloCuts_PHOS->GetXaxis()->SetBinLabel(3, "#it{M}_{02} cut");
+    hCaloCuts_PHOS->GetXaxis()->SetBinLabel(4, "#it{E} cut");
+    hCaloCuts_PHOS->GetXaxis()->SetBinLabel(5, "#it{N}_{cell} cut");
+    hCaloCuts_PHOS->GetXaxis()->SetBinLabel(6, "TM");
+    hCaloCuts_PHOS->GetXaxis()->SetBinLabel(7, "out");
+    if (PHOS_QA & 0b1) {
+      PHOSHistos.add("clusterTM_dEtadPhi", "cluster trackmatching dEta/dPhi;d#it{#eta};d#it{#varphi} (rad)", kTH2F, {{100, -0.2, 0.2}, {100, -0.2, 0.2}}); // dEta dPhi map of matched tracks
+    }
+
+    LOG(info) << "| PHOS cluster cut settings:" << std::endl;
+    LOG(info) << "|\t Timing cut: " << PHOS_minTime << " < t < " << PHOS_maxTime << std::endl;
+    LOG(info) << "|\t NCell cut: " << PHOS_minNCell << " <= NCell for E >= " << PHOS_minENCell << std::endl;
+    LOG(info) << "|\t M02 cut: " << PHOS_minM02 << " < M02 for E >= " << PHOS_minENCell << std::endl;
+    LOG(info) << "|\t E_min cut: E_cluster > " << PHOS_minE << std::endl;
+    LOG(info) << "|\t TM |eta|: |eta| <= " << PHOS_TM_Eta << std::endl;
+    LOG(info) << "|\t TM |phi|: |phi| <= " << PHOS_TM_Phi << std::endl;
+  }
+
+  void processRec(aod::EMReducedEvents const&, aod::SkimEMCClusters const& emcclusters, aod::SkimEMCMTs const& matchedtracks, aod::PHOSClusters const& phosclusters)
+  {
+    for (int iCut = 0; iCut < 64; iCut++) {                   // loop over max number of cut settings
```

Comment:
Did you think about changing the order of loops? 
Like an outer loop over clusters and an inner loop over iCut? Then you would have to do the `tracks.sliceBy(perEMCClusterMT, cluster.globalIndex());` in gammaSelectionCuts::doPhotonCutsEMC(...) only once per cluster. Unless Im overseeing sth.

Reply:
yes sounds good, also to follow the spirit of encapsulation in oop ^^